### DISCUSSION TITLE
`horizontalPanelAlignment`: Give us your thoughts!

### DISCUSSION THREAD

**Discussion by raplemie** [iTwin/appui#706] [Preview Features]
# Feature description

The top and bottom panels will now have an _Align_ button, which will enable the user to change the panel behavior, it will allow them to stick the top panel to the left (making the right panel all the way to the top of the view), right (left panel to the top) center (both left and right panel to the top) and justify (default behavior, top panel full width of the view, side panels snapping to the top panel) and similar to the bottom. This configuration is currently NOT saved anywhere but would be just like the position of the widgets if it was implemented in a complete feature.

The basis for this request was to allow more user control over the disposition when it would be useful for the user to have more vertical panel space than horizontal one (and in a 16:9 configuration, a user may prefer this)

> [!NOTE]  
> Initially released in 4.9.0.

![horizontalPanelAlignment](https://github.com/iTwin/appui/assets/849817/f9795bfa-4c94-4d0c-8dc5-a12931316c6a)

## Why is it a preview?

- The widget "buttons" area is getting crowded, so there may be other options.
- The interaction might need improvement and we prefer validating it before forcing it.
- We want to determine if the new behavior should be the default and only option, or rather if it should be an "opt-in" and require related API to be enabled (are there cases where you would NOT want a user to be able to control this)

## What then?

We would like the community to chime in on any issues they are facing when using this feature, and discuss how we can resolve these. We'd also like to know if there is a particular API that would be useful with this feature, additional options to be considered, modified or additional behavior, or any other ideas or opinions on the subject.

Basically, feel free to add comments on its usefulness and how this feature should be officially supported!

**Comment by VeroniqueVezina** [comment]
I am a bit unsure about this one. User customization is great, but the feeling I have is that this was requested because we do not offer a good default layout to product teams to start with.
